### PR TITLE
Add authentication for app.js events.js and api.js

### PR DESCRIPTION
**Pull request by rogerdunnhawaii**
-User will now be able to sign-up, sign-in, change pw, sign-out.
-Still have ui.js left to do
-Using promise and .then and .catch to send to ui.js

### CODE REVIEW

File: assets/scripts/auth/api.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const config = require('../config.js')
+const store = require('../store.js')
+
+const signUp = function (data) {
+  return $.ajax({
+    url: config.apiUrl + '/sign-up',
+    method: POST,
```
Comment (macitege) [inline assets/scripts/auth/api.js]: Methods should be in quotes.

File: assets/scripts/auth/api.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      Authorization: 'Token token=' + store.user.token
+    },
+    data: data
+  })
+}
+
+const signOut = function () {
+  return $.ajax({
+    url: config.apiUrl + '/sign-out',
+    method: DESTROY,
```
Comment (macitege) [inline assets/scripts/auth/api.js]: DESTROY should be DELETE

**Comment by macitege** [comment]
Merger

**Comment by rogerdunnhawaii** [comment]
shall I merge or wait on Maxx?

**Comment by discocisco** [comment]
Never merge your own pull request, I don't think we need to do more work tonight so it's fine to wait for Maxx

**Comment by MaxxKowalik** [comment]
Looks good! Lets merge

**Pull request merged**